src/praisonai-agents/praisonaiagents/guardrails/protocols.py

```
@@ -0,0 +1,247 @@
+        return self._validate_through_chain("validate_output", content, **kwargs)
+        
+    def validate_tool_call(self, tool_name: str, arguments: Dict[str, Any], **kwargs) -> Tuple[bool, Dict[str, Any]]:
+        """Validate tool call through all guardrails."""
+        for guardrail in self.guardrails:
+            if hasattr(guardrail, 'validate_tool_call'):
+                try:
+                    is_valid, processed_args = guardrail.validate_tool_call(tool_name, arguments, **kwargs)
+                    if not is_valid:
+                        return False, arguments  # Failed validation
+                    arguments = processed_args  # Update arguments with processing
+                except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)